viewing_party/party.py

```diff
@@ -1,23 +1,126 @@
+    if len(user_data["watched"]) == 0:
+        average_ratings = 0.0
+        return average_ratings
+    for movie in user_data["watched"]:
+        number_of_movies += 1
+        average_ratings += float(movie["rating"])
+    average_ratings = average_ratings/number_of_movies
+    return average_ratings
+        
+def get_most_watched_genre(user_data):
+# to include multiple genres that have high count
+    genre_dict = {}
+    most_frequent_genre = ""
+    if len(user_data["watched"]) == 0:
+        return None
+    for movie in user_data["watched"]:
+        if movie["genre"] not in genre_dict:
+            genre_dict[movie["genre"]] = 1
+        elif movie["genre"] in genre_dict:
+            genre_dict[movie["genre"]] += 1
+    most_frequent_genre = [key for key, value in genre_dict.items()\
+         if value == max(genre_dict.values())]
+    return ' '.join(most_frequent_genre)
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_user_watched_movies(user_data):
+    user_watched_movies = []
+    for movie in user_data["watched"]:
+        user_watched_movies.append(movie)
+    return user_watched_movies
+
+def get_friends_watched_movies(user_data):
+    friends_watched_movies = []
+    for friend in user_data["friends"]:
+        for film in friend["watched"]:
+            if film not in friends_watched_movies:
+                friends_watched_movies.append(film)
+    return friends_watched_movies
+
+def get_unique_watched(user_data):
+    user_watched_movies = get_user_watched_movies(user_data)
+    friends_watched_movies = get_friends_watched_movies(user_data)
+    unique_movies = []
+    for item in user_watched_movies:
+        if item not in friends_watched_movies:
+            unique_movies.append(item)
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+    user_watched_movies = get_user_watched_movies(user_data)
+    friends_watched_movies = get_friends_watched_movies(user_data)
+    unique_movies = []
+    for movie in friends_watched_movies:
+        if movie not in user_watched_movies:
+            unique_movies.append(movie)
+    return unique_movies
+
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    movie_recs = []
+    friends_watched_user_no = get_friends_unique_watched(user_data)
+    for movie in friends_watched_user_no:
+        if movie["host"] in user_data["subscriptions"]:
+            movie_recs.append(movie)
+    return movie_recs
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):
+    new_recs = []
+    user_most_watched_genre = get_most_watched_genre(user_data)
+    friends_watched_user_no = get_friends_unique_watched(user_data)

```

Comment:
What does 'no' in this variable refer to? Is it number?

Consider renaming `friends_watched_user_no` to something like friends_unique_watched_movies for a little more clarity.